app/_components/GuessDistanceModal.tsx

```diff
@@ -14,14 +17,67 @@ const GuessDistanceModal = ({
   setGuessPoiPosition: (arg0: Coordinates | null) => void;
   userCoordinates: Coordinates;
 }) => {
+  const importantPinContext = useContext(ImportantPinContext);
+  const [distanceToPin, setDistancePin] = useState<number>(0);
+  const [hint, setHint] = useState<string>(''); 
+  const drawerRef = useRef<HTMLButtonElement>(null); // Ref for the Done button
+  const thresholdDistance = 20;
+
+  useEffect(() => {
+    if (!importantPinContext) return;
+    if (!importantPinContext.guessedPin) return;
```

Comment:
We can combine this into one line, I think.
```suggestion
    if (!importantPinContext || !importantPinContext.guessedPin) return;
```